cclient/iota_client_extended_api.c

```diff
@@ -56,6 +56,108 @@ static retcode_t is_unused_address(iota_client_service_t const* const serv,
   return ret_code;
 }
 
+static retcode_t traverse_bundle(iota_client_service_t const* const serv,
+                                 flex_trit_t const* const tail_hash,
+                                 bundle_transactions_t* const bundle,
+                                 hash8019_array_p trytes) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_req_t* get_trytes_req = NULL;
+  get_trytes_res_t* get_trytes_res = NULL;
+  iota_transaction_t tx = {};
+  flex_trit_t* tmp_trytes = NULL;
+  flex_trit_t trunk_hash[FLEX_TRIT_SIZE_243];
+  int64_t current_index = 0, last_index = 0;
+
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+
+  get_trytes_req = get_trytes_req_new();
+  if (!get_trytes_req) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(logger_id, "%s get_trytes_req_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+  get_trytes_res = get_trytes_res_new();
+  if (!get_trytes_res) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(logger_id, "%s get_trytes_res_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+
+  memcpy(trunk_hash, tail_hash, FLEX_TRIT_SIZE_243);
+  do {
+    transaction_reset(&tx);
+    // Create request and response to get_trytes API
+    // Add the trunk hash to the request parameters
+    ret_code = hash243_queue_push(&get_trytes_req->hashes, trunk_hash);
+    if (ret_code != RC_OK) {
+      log_error(logger_id, "%s hash243_queue_push failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    // Call the get_trytes API
+    ret_code = iota_client_get_trytes(serv, get_trytes_req, get_trytes_res);
+    if (ret_code != RC_OK) {
+      log_error(logger_id, "%s iota_client_get_trytes failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    hash243_queue_free(&get_trytes_req->hashes);
+    // Get the transaction trytes
+    tmp_trytes = hash8019_queue_at(&get_trytes_res->trytes, 0);
+    if (!tmp_trytes) {
+      ret_code = RC_CCLIENT_RES_ERROR;
+      log_error(logger_id, "%s hash8019_queue_at failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    // Create a new transaction with the received trytes
+    transaction_deserialize_from_trits(&tx, tmp_trytes, true);
+
+    current_index = transaction_current_index(&tx);
+    last_index = transaction_last_index(&tx);
+    // Get the next tail_hash from the transaction
+    memcpy(trunk_hash, transaction_trunk(&tx), FLEX_TRIT_SIZE_243);
```

Comment:
Can't `trunk_hash` just be a pointer to avoid copies ?

Reply:
I don't think so.